### PR TITLE
br/lightning: add cleanUp files util

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #45719

Problem Summary:

### What is changed and how it works?
CleanUp all files under one subdir.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/external/util_test.go [hunk 1]
```diff
@@ -171,3 +171,40 @@ func TestGetAllFileNames(t *testing.T) {
...
 }
+
+func TestCleanUpFiles(t *testing.T) {
+	ctx := context.Background()
+	store := storage.NewMemStorage()
+	w := NewWriterBuilder().
+		SetMemorySizeLimit(20).
+		SetPropSizeDistance(5).
+		SetPropKeysDistance(3).
+		Build(store, 0, "/subtask")
```
Comment (tangenta) [inline br/pkg/lightning/backend/external/util_test.go:182]:
```suggestion
		Build(store, "/subtask", 0)
```

I have changed the parameter order in #46387.

**Review by tangenta — APPROVED** [review]

File: br/pkg/lightning/backend/external/util.go [hunk 2]
```diff
@@ -118,3 +118,26 @@ func GetAllFileNames(
...
+
+// CleanUpFiles delete all data and stat files under one subDir.
+func CleanUpFiles(ctx context.Context,
+	store storage.ExternalStorage,
+	subDir string) error {
+	dataNames, statNames, err := GetAllFileNames(ctx, store, subDir)
+	if err != nil {
+		return err
+	}
+	for _, data := range dataNames {
```
Comment (lance6716) [inline br/pkg/lightning/backend/external/util.go:130]: maybe delete them concurrently considering network overhead

**Review by wjhuang2016 — APPROVED** [review]

**Comment by lance6716** [comment]
/hold

need to address comment

File: br/pkg/lightning/backend/external/util.go [hunk 3]
```diff
@@ -118,3 +121,39 @@ func GetAllFileNames(
...
+	if err != nil {
+		return err
+	}
+
+	eg, _ := errgroup.WithContext(ctx)
+	workerPool := utils.NewWorkerPool(concurrency, "delete global sort files")
+	for i := range dataNames {
+		data := dataNames[i]
+		workerPool.ApplyOnErrorGroup(eg, func() error {
+			logutil.BgLogger().Info("data", zap.Any("data", data))
```
Comment (lance6716) [inline br/pkg/lightning/backend/external/util.go:140]: can try to use the logger in `ctx` by `logutil.Logger(ctx)`

File: br/pkg/lightning/backend/external/util.go [hunk 4]
```diff
@@ -118,3 +121,39 @@ func GetAllFileNames(
...
+func CleanUpFiles(ctx context.Context,
+	store storage.ExternalStorage,
+	subDir string,
+	concurrency uint) error {
+	dataNames, statNames, err := GetAllFileNames(ctx, store, subDir)
+	if err != nil {
+		return err
+	}
+
+	eg, _ := errgroup.WithContext(ctx)
```
Comment (lance6716) [inline br/pkg/lightning/backend/external/util.go:135]: if we don't need to check the error from eg.Go, we can use the zero value of errgroup.Group

**Pull request merged**